mesonbuild/modules/gnome.py

```diff
@@ -328,16 +328,17 @@ def _get_dependencies_flags(self, deps, state, depends, include_rpath=False,
             if isinstance(dep, InternalDependency):
                 cflags.update(get_include_args(dep.include_directories))
                 for lib in dep.libraries:
-                    if hasattr(lib, 'held_object'):
-                        lib = lib.held_object
-                    internal_ldflags.update(self._get_link_args(state, lib, depends, include_rpath))
-                    libdepflags = self._get_dependencies_flags(lib.get_external_deps(), state, depends, include_rpath,
-                                                               use_gir_args, True)
-                    cflags.update(libdepflags[0])
-                    internal_ldflags.update(libdepflags[1])
-                    external_ldflags.update(libdepflags[2])
-                    external_ldflags_nodedup += libdepflags[3]
-                    gi_includes.update(libdepflags[4])
+                    if isinstance(lib, build.SharedLibrary):
+                        if hasattr(lib, 'held_object'):
```

Comment:
This can not happen. If `lib` does have a `held_object` then it is a `LibraryHolder` and thus the `isinstance` check above will fail. These tests need to be in the opposite order.